### PR TITLE
Add bbe for converting csv string to `anydata[][]`

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes https://github.com/ballerina-platform/ballerina-dev-website/issues/9449

### CODE REVIEW

File: examples/csv-string-to-anydata-array/csv_string_to_anydata_array.bal [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import ballerina/data.csv;
+import ballerina/io;
+
+public function main() {
+    string csvString = string `name,author,price,publishedDate
+                               Effective Java,Joshua Bloch,45.99,2018-01-01
+                               Clean Code,Robert C. Martin,37.50,2008-08-01`;
+
+    // Parse the CSV string to a array of string arrays.
```
Comment (MaryamZi) [inline examples/csv-string-to-anydata-array/csv_string_to_anydata_array.bal:9]:
```suggestion
    // Parse the CSV string as an array of string arrays.
```

File: examples/csv-string-to-anydata-array/csv_string_to_anydata_array.bal [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    string[][]|csv:Error bookArray = csv:parseString(csvString);
+    io:println(bookArray);
+
+    // Parse the CSV string to a array of anydata arrays.
+    anydata[][]|csv:Error bookArray2 = csv:parseString(csvString);
+    io:println(bookArray2);
+
+    // Parse the CSV string to a array of string arrays with data projection.
```
Comment (MaryamZi) [inline examples/csv-string-to-anydata-array/csv_string_to_anydata_array.bal:17]: Fix as suggested for previous comments.

File: examples/csv-string-to-anydata-array/csv_string_to_anydata_array.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Convert CSV string to array of anydata arrays
```
Comment (MaryamZi) [inline examples/csv-string-to-anydata-array/csv_string_to_anydata_array.md:1]: Convert?

File: examples/csv-string-to-anydata-array/csv_string_to_anydata_array.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+The Ballerina `data.csv` library offers a variety of functions for converting CSV strings to sub types of `anydata[][]`, enabling developers to effectively parse CSV string data to flexible data structures. Additionally, it provides the option to select specific fields for inclusion in the resulting array of anydata arrays, ensuring that developers can tailor the output to meet their application's needs
```
Comment (MaryamZi) [inline examples/csv-string-to-anydata-array/csv_string_to_anydata_array.md:3]: - offers a variety of functions -> offers a number of functions
- please fix convert/parse
- just say arrays in the heading
- use backticks where required
- why is there no full stop at the end?
- ensuring that developers can tailor the output to meet their application's needs - I would be more specific to say extract only the required data

File: examples/csv-string-to-anydata-array/csv_string_to_anydata_array.metatags [hunk 5]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates conversion of CSV string to a Ballerina array of anydata arrays with and without data projection.
```
Comment (MaryamZi) [inline examples/csv-string-to-anydata-array/csv_string_to_anydata_array.metatags:1]: - grammar issues
- parse/convert

File: examples/csv-string-to-anydata-array/csv_string_to_anydata_array.metatags [hunk 6]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates the parsing of a CSV string into an array of anydata arrays, both with and without data projection.
+keywords: ballerina, ballerina by example, bbe, csv, csv string, anydata, anydata array, parseString, csv data module, data.csv, data projection
```
Comment (SasinduDilshara) [inline examples/csv-string-to-anydata-array/csv_string_to_anydata_array.metatags:2]: Add `csv to array` keyword

**Review by gimantha — APPROVED** [review]

**Pull request merged**